Horizon/Horizon/Sources/Features/Dashboard/View/DashboardView.swift

```diff
@@ -50,53 +51,17 @@ struct DashboardView: View {
                 }
             }
         }
-        .navigationBarItems(leading: nameLabel)
-        .navigationBarItems(trailing: navBarIcons)
+        .navigationBarItems(leading: HorizonUI.NavigationBar.Leading(title: "Hi, Horizon App"))
+        .navigationBarItems(trailing: HorizonUI.NavigationBar.Trailing(onEvent: viewModel.onEvent))
         .scrollIndicators(.hidden, axes: .vertical)
         .background(Color.backgroundLight)
+        .onFirstAppear { viewModel.viewController =  viewController}
```

Comment:
I don't love the pattern of requiring the view to set a property on the view model. I think it's better just to pass the viewController as an argument to the notebookDidTap  method. That way it's always clear what is required. Otherwise it can be unclear to someone using the view model that they need to set that property before it'll work.

Reply:
Ok but i need to pass the closure direct to view model, because the navigation logic  must be set in viewModel.

Reply:
It would just be one of the parameters passed in the onEvent handler:

func onEvent(event: HorizonUI.NavigationBar.Trailing.Event, viewController: WeakViewController)

You might also consider creating a separate ViewModel to pair with the HorizonUI.NavigationBar.Trailing and putting the onEvent handler in there. Otherwise we must handle the navigation logic in the ViewModel for every View that uses the HorizonUI.NavigationBar.Trailing view.

Reply:
No, HorizonUI is just for creating UI only and shouldn't have any logic in it.